core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -78,9 +79,18 @@
      */
     public final void initialize(TelemetryConfiguration configuration) {
         try {
-            ApplicationInsightsXmlConfiguration applicationInsights = builder.build(getConfigurationFileName());
+            InternalLogger.INSTANCE.configMode(true);
+
+            String configurationFileName = getConfigurationFileName();
+            if (Strings.isNullOrEmpty(configurationFileName)) {
+                return;
+            }
+
+            ApplicationInsightsXmlConfiguration applicationInsights = builder.build(configurationFileName);
```

Comment:
Please rename the variable name to reflect that it is a configuration, i.e. `applicationInsightsConfig`.